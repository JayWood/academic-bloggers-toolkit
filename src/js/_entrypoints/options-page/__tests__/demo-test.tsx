import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Demo from '../demo';

const setup = ({
    bib_heading = 'Bibliography',
    bib_heading_level = 'h3',
    bibliography = 'fixed',
    links = 'always',
}: Partial<ABT.DisplayOptions> = {}) => {
    const options = {
        bib_heading,
        bib_heading_level,
        bibliography,
        links,
    };
    const component = shallow(<Demo options={options} />);
    return {
        component,
    };
};

describe('<Demo />', () => {
    describe('Snapshots', () => {
        it('should render headings', () => {
            const { component } = setup();
            const BASE = toJSON(component);
            expect(BASE).toMatchSnapshot();

            const { component: headingChanged } = setup({
                bib_heading: 'Testing',
            });
            expect(toJSON(headingChanged)).toMatchDiffSnapshot(BASE);

            const { component: withoutHeading } = setup({ bib_heading: '' });
            expect(toJSON(withoutHeading)).toMatchDiffSnapshot(BASE);
        });
        it('should render correct heading level', () => {
            const { component } = setup();
            const BASE = toJSON(component);
            expect(BASE).toMatchSnapshot();

            for (const bib_heading_level of [
                'h1',
                'h2',
                'h3',
                'h4',
                'h5',
                'h6',
            ] as ABT.HeadingLevel[]) {
                const { component: levelChanged } = setup({
                    bib_heading_level,
                });
                expect(toJSON(levelChanged)).toMatchDiffSnapshot(BASE);
            }
        });
        it('should render correct link format', () => {
            const { component } = setup();
            const BASE = toJSON(component);
            expect(BASE).toMatchSnapshot();

            const { component: alwaysFullSurround } = setup({
                links: 'always-full-surround',
            });
            expect(toJSON(alwaysFullSurround)).toMatchDiffSnapshot(BASE);

            const { component: urls } = setup({ links: 'urls' });
            expect(toJSON(urls)).toMatchDiffSnapshot(BASE);

            const { component: never } = setup({ links: 'never' });
            expect(toJSON(never)).toMatchDiffSnapshot(BASE);
        });
    });
    it('should handle click', () => {
        const { component } = setup();
        const heading = component.find('h3').first();
        expect(component.state().isToggled).toBe(false);
        heading.simulate('click');
        expect(component.state().isToggled).toBe(true);
    });
    it('should handle prop change', () => {
        const options: ABT.DisplayOptions = {
            bib_heading: 'Bibliography',
            bib_heading_level: 'h3',
            links: 'always',
            bibliography: 'toggle',
        };
        const { component } = setup(options);
        component.setState({ isToggled: true });

        // `bibliography` prop idential. No state change
        component.setProps({ options });
        expect(component.state().isToggled).toBe(true);

        // `bibliography` prop different. State change.
        component.setProps({ options: { ...options, bibliography: 'fixed' } });
        expect(component.state().isToggled).toBe(false);
    });
});
