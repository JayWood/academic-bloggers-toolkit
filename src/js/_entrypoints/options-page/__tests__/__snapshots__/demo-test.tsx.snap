// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Demo /> Snapshots should render correct heading level 1`] = `
<div
  className="demo"
>
  <h3
    aria-controls="bibliography-content"
    className="demoHeading"
    onClick={[Function]}
    role="button"
    style={
      Object {
        "fontSize": "1.8em",
        "lineHeight": "1.8em",
      }
    }
  >
    Bibliography
  </h3>
  <div
    aria-hidden={false}
    id="bibliography-content"
    style={
      Object {
        "display": "initial",
      }
    }
  >
    <span
      className="demoNumber"
    >
      1.
    </span>
    <span>
      Sifford D P. Academic Blogger’s Toolkit:
       
      <a
        href="https://wordpress.org/plugins/academic-bloggers-toolkit/"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://wordpress.org/plugins/academic-bloggers-toolkit/
      </a>
      .
      <i>
         J WordPress
      </i>
      . 2015;12(5):12-24. [
      <a
        href="https://dx.doi.org/#1"
        rel="noopener noreferrer"
        target="_blank"
      >
        Source
      </a>
      ]
    </span>
  </div>
</div>
`;

exports[`<Demo /> Snapshots should render correct heading level 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,12 +6,12 @@
      className=\\"demoHeading\\"
      onClick={[Function anonymous]}
      role=\\"button\\"
      style={
        Object {
-         \\"fontSize\\": \\"2.5em\\",
-         \\"lineHeight\\": \\"2.5em\\",
+         \\"fontSize\\": \\"1.8em\\",
+         \\"lineHeight\\": \\"1.8em\\",
        }
      }
    >
      Bibliography
    </h3>"
`;

exports[`<Demo /> Snapshots should render correct heading level 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,12 +6,12 @@
      className=\\"demoHeading\\"
      onClick={[Function anonymous]}
      role=\\"button\\"
      style={
        Object {
-         \\"fontSize\\": \\"2.2em\\",
-         \\"lineHeight\\": \\"2.2em\\",
+         \\"fontSize\\": \\"1.8em\\",
+         \\"lineHeight\\": \\"1.8em\\",
        }
      }
    >
      Bibliography
    </h3>"
`;

exports[`<Demo /> Snapshots should render correct heading level 4`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`<Demo /> Snapshots should render correct heading level 5`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,12 +6,12 @@
      className=\\"demoHeading\\"
      onClick={[Function anonymous]}
      role=\\"button\\"
      style={
        Object {
-         \\"fontSize\\": \\"1.4em\\",
-         \\"lineHeight\\": \\"1.4em\\",
+         \\"fontSize\\": \\"1.8em\\",
+         \\"lineHeight\\": \\"1.8em\\",
        }
      }
    >
      Bibliography
    </h3>"
`;

exports[`<Demo /> Snapshots should render correct heading level 6`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,12 +6,12 @@
      className=\\"demoHeading\\"
      onClick={[Function anonymous]}
      role=\\"button\\"
      style={
        Object {
-         \\"fontSize\\": \\"1.2em\\",
-         \\"lineHeight\\": \\"1.2em\\",
+         \\"fontSize\\": \\"1.8em\\",
+         \\"lineHeight\\": \\"1.8em\\",
        }
      }
    >
      Bibliography
    </h3>"
`;

exports[`<Demo /> Snapshots should render correct heading level 7`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,12 +6,12 @@
      className=\\"demoHeading\\"
      onClick={[Function anonymous]}
      role=\\"button\\"
      style={
        Object {
-         \\"fontSize\\": \\"1.0em\\",
-         \\"lineHeight\\": \\"1.0em\\",
+         \\"fontSize\\": \\"1.8em\\",
+         \\"lineHeight\\": \\"1.8em\\",
        }
      }
    >
      Bibliography
    </h3>"
`;

exports[`<Demo /> Snapshots should render correct link format 1`] = `
<div
  className="demo"
>
  <h3
    aria-controls="bibliography-content"
    className="demoHeading"
    onClick={[Function]}
    role="button"
    style={
      Object {
        "fontSize": "1.8em",
        "lineHeight": "1.8em",
      }
    }
  >
    Bibliography
  </h3>
  <div
    aria-hidden={false}
    id="bibliography-content"
    style={
      Object {
        "display": "initial",
      }
    }
  >
    <span
      className="demoNumber"
    >
      1.
    </span>
    <span>
      Sifford D P. Academic Blogger’s Toolkit:
       
      <a
        href="https://wordpress.org/plugins/academic-bloggers-toolkit/"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://wordpress.org/plugins/academic-bloggers-toolkit/
      </a>
      .
      <i>
         J WordPress
      </i>
      . 2015;12(5):12-24. [
      <a
        href="https://dx.doi.org/#1"
        rel="noopener noreferrer"
        target="_blank"
      >
        Source
      </a>
      ]
    </span>
  </div>
</div>
`;

exports[`<Demo /> Snapshots should render correct link format 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,21 +28,30 @@
        className=\\"demoNumber\\"
      >
        1.
      </span>
      <span>
+       Sifford D P. Academic Blogger’s Toolkit:
+        
        <a
-         href=\\"https://dx.doi.org/#2\\"
+         href=\\"https://wordpress.org/plugins/academic-bloggers-toolkit/\\"
          rel=\\"noopener noreferrer\\"
          target=\\"_blank\\"
        >
-         Sifford D P. Academic Blogger’s Toolkit:
-          
-         https://wordpress.org/plugins/academic-bloggers-toolkit/.
+         https://wordpress.org/plugins/academic-bloggers-toolkit/
+       </a>
+       .
        <i>
           J WordPress
        </i>
-         . 2015;12(5):12-24.
+       . 2015;12(5):12-24. [
+       <a
+         href=\\"https://dx.doi.org/#1\\"
+         rel=\\"noopener noreferrer\\"
+         target=\\"_blank\\"
+       >
+         Source
        </a>
+       ]
      </span>
    </div>
  </div>"
`;

exports[`<Demo /> Snapshots should render correct link format 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,21 +28,30 @@
        className=\\"demoNumber\\"
      >
        1.
      </span>
      <span>
-       Sifford D P. Academic Blogger’s Toolkit: 
+       Sifford D P. Academic Blogger’s Toolkit:
+        
        <a
          href=\\"https://wordpress.org/plugins/academic-bloggers-toolkit/\\"
          rel=\\"noopener noreferrer\\"
          target=\\"_blank\\"
        >
          https://wordpress.org/plugins/academic-bloggers-toolkit/
        </a>
-       . 
+       .
        <i>
-         J WordPress
+          J WordPress
        </i>
-       . 2015;12(5):12-24.
+       . 2015;12(5):12-24. [
+       <a
+         href=\\"https://dx.doi.org/#1\\"
+         rel=\\"noopener noreferrer\\"
+         target=\\"_blank\\"
+       >
+         Source
+       </a>
+       ]
      </span>
    </div>
  </div>"
`;

exports[`<Demo /> Snapshots should render correct link format 4`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -30,14 +30,28 @@
        1.
      </span>
      <span>
        Sifford D P. Academic Blogger’s Toolkit:
         
+       <a
+         href=\\"https://wordpress.org/plugins/academic-bloggers-toolkit/\\"
+         rel=\\"noopener noreferrer\\"
+         target=\\"_blank\\"
+       >
          https://wordpress.org/plugins/academic-bloggers-toolkit/
-       . 
+       </a>
+       .
        <i>
-         J WordPress
+          J WordPress
        </i>
-       . 2015;12(5):12-24.
+       . 2015;12(5):12-24. [
+       <a
+         href=\\"https://dx.doi.org/#1\\"
+         rel=\\"noopener noreferrer\\"
+         target=\\"_blank\\"
+       >
+         Source
+       </a>
+       ]
      </span>
    </div>
  </div>"
`;

exports[`<Demo /> Snapshots should render headings 1`] = `
<div
  className="demo"
>
  <h3
    aria-controls="bibliography-content"
    className="demoHeading"
    onClick={[Function]}
    role="button"
    style={
      Object {
        "fontSize": "1.8em",
        "lineHeight": "1.8em",
      }
    }
  >
    Bibliography
  </h3>
  <div
    aria-hidden={false}
    id="bibliography-content"
    style={
      Object {
        "display": "initial",
      }
    }
  >
    <span
      className="demoNumber"
    >
      1.
    </span>
    <span>
      Sifford D P. Academic Blogger’s Toolkit:
       
      <a
        href="https://wordpress.org/plugins/academic-bloggers-toolkit/"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://wordpress.org/plugins/academic-bloggers-toolkit/
      </a>
      .
      <i>
         J WordPress
      </i>
      . 2015;12(5):12-24. [
      <a
        href="https://dx.doi.org/#1"
        rel="noopener noreferrer"
        target="_blank"
      >
        Source
      </a>
      ]
    </span>
  </div>
</div>
`;

exports[`<Demo /> Snapshots should render headings 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -11,11 +11,11 @@
          \\"fontSize\\": \\"1.8em\\",
          \\"lineHeight\\": \\"1.8em\\",
        }
      }
    >
-     Testing
+     Bibliography
    </h3>
    <div
      aria-hidden={false}
      id=\\"bibliography-content\\"
      style={"
`;

exports[`<Demo /> Snapshots should render headings 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,22 @@
  <div
    className=\\"demo\\"
  >
+   <h3
+     aria-controls=\\"bibliography-content\\"
+     className=\\"demoHeading\\"
+     onClick={[Function anonymous]}
+     role=\\"button\\"
+     style={
+       Object {
+         \\"fontSize\\": \\"1.8em\\",
+         \\"lineHeight\\": \\"1.8em\\",
+       }
+     }
+   >
+     Bibliography
+   </h3>
    <div
      aria-hidden={false}
      id=\\"bibliography-content\\"
      style={
        Object {"
`;
